k6/endpoints/products.js

```diff
@@ -0,0 +1,49 @@
+// run k6 run k6/endpoints/products.js
+// summary file uploaded at k6/summary/products.html
+import http from 'k6/http';
+import { sleep, check } from 'k6';
+import { Counter } from 'k6/metrics';
+import { htmlReport } from "https://raw.githubusercontent.com/benc-uk/k6-reporter/main/dist/bundle.js";
+
+// allows for incrementation of counter for every error
+const errors = new Counter('errors');
+
+// defines test run behaviors
+export const options = {
+  stages: [
+    { duration: '30s', target: 1500 }, // manually ramping the VUs to target of 2200
+    { duration: '30s', target: 2200 }
+  ],
+
+  thresholds: {
+    http_req_failed: ['rate < 0.01'], // http errors should be less than 1%
+    http_req_duration: ['avg < 2000'] // average latency rate should be less than 2000ms
+  }
+};
+
+// generates the summary report in UI
+export function handleSummary(data) {
+  return {
+    "k6/summary/products.html": htmlReport(data)
+  }
+};
+
+// iteration that each VU runs through
+export default () => {
+  // generates a random product id every iteration
+  //
+  const product_id = Math.floor(Math.random() * 1000010) + 1;
+  const res = http.get(`http://localhost:3000/products/${product_id}`);
```

Comment:
Just curious - why did you create a separate test for each endpoint along with a test for all 3?

Reply:
I created a separate test for each endpoint to see which one had the greatest latency rate. Basically to determine which endpoint is the slowest.